tensornetwork/matrixproductstates/mps.py

```diff
@@ -0,0 +1,564 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from tensornetwork.network import TensorNetwork
+import numpy as np
+from typing import Any, List, Optional, Text, Type, Union, Dict
+Tensor = Any
+
+
+class FiniteMPS(TensorNetwork):
+  """
+  An MPS class for finite systems.
+  `FiniteMPS` keeps track of the nodes of the network by storing them in a list
+  `FiniteMPS.nodes`. Any external changes to this list will potentially corrupt 
```

Comment:
You can make that a property to avoid any modifications. Sure, someone could always modify `FiniteMPS._nodes` too, but then that is their own fault. :)